lib/assert.js

```diff
@@ -958,5 +959,100 @@ export class Assertions {
 			pass();
 			return true;
 		});
+
+		this.unorderedEqual = withSkip((actual, expected, message) => {
+			if (!checkMessage('unorderedEqual', message)) {
+				return false;
+			}
+
+			const actualInfo = checkValueForUnorderedEqual(actual);
+
+			if (!actualInfo.isValid) {
+				fail(new AssertionError({
+					assertion: 'unorderedEqual',
+					improperUsage: true,
+					message: '`t.unorderedEqual` only compares Maps, Sets, and arrays',
+					values: [formatWithLabel('Called with:', actual)],
+				}));
+				return false;
+			}
+
+			const expectedInfo = checkValueForUnorderedEqual(expected);
+
+			if (!expectedInfo.isValid) {
+				fail(new AssertionError({
+					assertion: 'unorderedEqual',
+					improperUsage: true,
+					message: '`t.unorderedEqual` only compares Maps, Sets, and arrays',
+					values: [formatWithLabel('Called with:', expected)],
+				}));
+				return false;
+			}
+
+			if (actualInfo.size !== expectedInfo.size) {
+				fail(new AssertionError({
+					assertion: 'unorderedEqual',
+					message: 'size must be equal',
+				}));
+				return false;
+			}
+
+			if (actualInfo.type === 'map') {
+				if (expectedInfo.type !== 'map') {
+					fail(new AssertionError({
+						assertion: 'unorderedEqual',
+						message: 'both must be maps',
+					}));
+					return false;
+				}
+
+				const comparedKeysResult = concordance.compare(actual.keys, expected.keys, concordanceOptions);
```

Comment:
Did you mean:

```suggestion
				const comparedKeysResult = concordance.compare(actual.keys(), expected.keys(), concordanceOptions);
```

Of course that returns an iterator, which Concordance may walk over (can't recall), but the keys won't be in the same order.

If both values have the same size, then you probably don't need this?

Reply:
Agh - the joys of plain JavaScript.

> If both values have the same size, then you probably don't need this?

Good catch, I think you're right as well.